### PR TITLE
Ensure buildpack versions are written to app image metadata

### PR DESCRIPTION
**Pull request by ameyer-pivotal**
Resolves issue #85

### CODE REVIEW

File: detector.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func mergeTOML(l *log.Logger, out io.Writer, in ...io.Reader) {
 
 type BuildpackOrder []BuildpackGroup
 
-func (bo BuildpackOrder) Detect(c *DetectConfig) (plan []byte, group *BuildpackGroup) {
+func (bo BuildpackOrder) Detect(c *DetectConfig) ([]byte, *BuildpackGroup) {
```
Comment (sclevine) [inline detector.go]: Keep named parameters for better documentation.

File: map_test.go [hunk 2]
```diff
@@ -85,9 +85,9 @@ func testMap(t *testing.T, when spec.G, it spec.S) {
 
 		it("should return an ordering of buildpacks", func() {
 			m := lifecycle.BuildpackMap{
-				"buildpack1@version1.1": {Name: "buildpack1-1.1"},
-				"buildpack1@version1.2": {Name: "buildpack1-1.2"},
-				"buildpack2@latest":     {Name: "buildpack2"},
+				"buildpack1@version1.1": {Name: "buildpack1-1.1", Version: "1.1"},
```
Comment (sclevine) [inline map_test.go]: What does this change accomplish?

File: detector.go (hunk 1 above)
Comment (joaopapereira) [reply detector.go]: We removed them because they were showing as obfuscated in the debugged

File: map_test.go (hunk 2 above)
Comment (joaopapereira) [reply map_test.go]: Ensure the version is being passed to the output.

Comment (sclevine) [reply map_test.go]: The returned objects are just copies of the original objects (see `lookup`), so the name seemed sufficient. No strong opinion, but we might want to make the same changes in the `#ReadGroup` test.

File: detector.go (hunk 1 above)
Comment (sclevine) [reply detector.go]: I think replacing `plan` and `group` on 202 with `p` and `g` would solve this and keep the godoc more clear, but approving either way.

Comment (ameyer-pivotal) [reply detector.go]: Ah yeah, I agree with @sclevine thinking about the gdocs. Optionally we can just call `return` but that never felt very readable.

File: map_test.go (hunk 2 above)
Comment (ameyer-pivotal) [reply map_test.go]: I guess I see your point. This change was simply one we chose to keep as we were writing tests to triage the bug. I'm OK either way.

Comment (sclevine) [reply map_test.go]: Would prefer to remove them then, to keep the test clear and consistent.

**Pull request merged**